load-test-ggs-git/index.js

```diff
@@ -0,0 +1,128 @@
+// This initialises a single simpleGit client that is re-used
+require("dotenv").config();
+const fs = require("fs").promises;
+const { Octokit } = require("@octokit/rest");
+const simpleGit = require("simple-git");
+
+const DIRECTORY_PATH = "./testDirectories";
+const REPO_PREFIX = "testRepo_";
+const TOTAL_REPOS = 200;
+const MAX_CONCURRENT_PROCESSES = 20;
+const RANDOM_OPS = 10;
+let successCount = 0;
+let failedCount = 0;
+
+const octokit = new Octokit({
+  auth: process.env.GITHUB_TOKEN,
+});
+
+const git = simpleGit({ maxConcurrentProcesses: MAX_CONCURRENT_PROCESSES });
+
+async function createRepoOngithub(repoName) {
+  return await octokit.repos.createForAuthenticatedUser({
+    name: repoName,
+    private: true,
+  });
+}
+
+async function logError(error) {
+  await fs.appendFile(
+    "errors.log",
+    `${error} at ${new Date().toISOString()}\n`
+  );
+}
+
+async function logInfo(info) {
+  await fs.appendFile("info.log", `${info} at ${new Date().toISOString()}\n`);
+}
+
+async function initializeRepo(i, createOnGithub) {
+  const repoDir = `${DIRECTORY_PATH}/${REPO_PREFIX}${i}`;
+
+  // Create directory and dummy file
+  await fs.mkdir(repoDir);
+  await logInfo(`Directory ${repoDir} created.`);
+
+  const randomContent = `Random content ${Math.random()}`;
+  await fs.writeFile(`${repoDir}/test.md`, randomContent);
+  await logInfo(`test.md created in ${repoDir} with random content at time.`);
+
+  // Initialize git with cwd() method
+  await git.cwd(repoDir).init();
```

Comment:
Sanity check here: Will the `master` branch be created here automatically? Cos from what I see there is no mention that the repo will be created with the `master` branch

Reply:
@dcshzj it will default to `main` branch so I didn't specify explicitly